Src/VimWpf/Implementation/BlockCaret/BlockCaret.cs

```diff
@@ -94,59 +104,64 @@ public double CaretOpacity
             }
         }
 
-        private ITextViewLine TextViewLineContainingCaret
+        private ITextViewLine GetTextViewLineContainingPoint(VirtualSnapshotPoint caretPoint)
         {
-            get
+            try
             {
-                try
+                if (!_textView.IsClosed && !_textView.InLayout)
                 {
-                    if (!_textView.IsClosed && !_textView.InLayout)
+                    var textViewLines = _textView.TextViewLines;
+                    if (textViewLines != null && textViewLines.IsValid)
                     {
-                        var caretPoint = _textView.Caret.Position.BufferPosition;
-                        var textViewLines = _textView.TextViewLines;
-                        if (textViewLines != null && textViewLines.IsValid)
+                        var line = textViewLines.GetTextViewLineContainingBufferPosition(caretPoint.Position);
+                        if (line != null && line.IsValid)
                         {
-                            var line = textViewLines.GetTextViewLineContainingBufferPosition(caretPoint);
-                            if (line != null && line.IsValid)
-                            {
-                                return line;
-                            }
-
+                            return line;
                         }
+
                     }
                 }
-                catch (InvalidOperationException ex)
-                {
-                    VimTrace.TraceError(ex);
-                }
-                return null;
             }
+            catch (InvalidOperationException ex)
+            {
+                VimTrace.TraceError(ex);
+            }
+            return null;
         }
 
         /// <summary>
         /// Is the real caret visible in some way?
         /// </summary>
-        private bool IsRealCaretVisible
+        private bool IsRealCaretVisible(VirtualSnapshotPoint caretPoint, out ITextViewLine textViewLine)
         {
-            get
+            textViewLine = null;
+            if (_textView.HasAggregateFocus)
             {
-                if (_textView.HasAggregateFocus)
+                textViewLine = GetTextViewLineContainingPoint(caretPoint);
+                if (textViewLine != null && textViewLine.VisibilityState != VisibilityState.Unattached)
                 {
-                    var line = TextViewLineContainingCaret;
-                    return line != null && line.VisibilityState != VisibilityState.Unattached;
+                    return true;
                 }
-                return false;
+                textViewLine = null;
             }
+            return false;
+        }
+
+        private bool IsRealCaretVisible(VirtualSnapshotPoint caretPoint)
+        {
+            return IsRealCaretVisible(caretPoint, out var textViewLine);

```

Comment:
Nit: given that `textViewLine` is unused here you could use a discard to make it explicity:

```
return IsRealCaretVisible(caretPoint, out _);
```